rskj-core/src/test/java/co/rsk/peg/feeperkb/FeePerKbSupportImplTest.java

```diff
@@ -0,0 +1,238 @@
+package co.rsk.peg.feeperkb;
+
+import static org.junit.jupiter.api.Assertions.*;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.Mockito.doNothing;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.never;
+import static org.mockito.Mockito.spy;
+import static org.mockito.Mockito.times;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+
+import co.rsk.bitcoinj.core.Coin;
+import co.rsk.peg.BridgeSerializationUtils;
+import co.rsk.peg.feeperkb.constants.FeePerKbMainNetConstants;
+import co.rsk.peg.vote.ABICallElection;
+import co.rsk.peg.vote.ABICallSpec;
+import co.rsk.peg.vote.AddressBasedAuthorizer;
+import java.util.Objects;
+import java.util.Optional;
+import org.ethereum.core.SignatureCache;
+import org.ethereum.core.Transaction;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.junit.jupiter.MockitoExtension;
+import org.mockito.junit.jupiter.MockitoSettings;
+import org.mockito.quality.Strictness;
+
+@ExtendWith(MockitoExtension.class)
+@MockitoSettings(strictness = Strictness.LENIENT)
+class FeePerKbSupportImplTest {
+
+    @Mock
+    FeePerKbStorageProvider provider;
+    @Mock
+    FeePerKbMainNetConstants feePerKbConstants;
+    @InjectMocks
+    FeePerKbSupportImpl feePerKbSupport;
+
+
+    @Test
+    void getFeePerKb() {
+        Optional<Coin> currentFeePerKb = Optional.of(Coin.COIN);
+        when(provider.getFeePerKb()).thenReturn(currentFeePerKb);
+
+        Coin actualResult = feePerKbSupport.getFeePerKb();
+
+        Coin expectedResult = currentFeePerKb.get();
+        assertEquals(expectedResult, actualResult);
+    }
+
+    @Test
+    void getFeePerKb_nullInStorageProvider() {
+        Optional<Coin> genesisFeePerKb = Optional.of(Coin.MILLICOIN.multiply(5));
+        when(provider.getFeePerKb()).thenReturn(genesisFeePerKb);
+
+        Coin actualResult = feePerKbSupport.getFeePerKb();
+
+        Coin expectedResult = genesisFeePerKb.get();
+        assertEquals(expectedResult, actualResult);
+    }
+
+    @Test
+    void voteFeePerKbChange_unsuccessfulVote_unauthorized() {
+        AddressBasedAuthorizer authorizer = mock(AddressBasedAuthorizer.class);
```

Comment:
This can also be created as an object instead of a mock. I can help you with this